nixos/modules/services/hardware/display.nix

```diff
@@ -167,13 +189,60 @@ in
         ++
         # selecting EDID for displays
         lib.trivial.pipe cfg.outputs [
-          (lib.attrsets.filterAttrs (_: spec: spec.edid != null))
+          (lib.attrsets.filterAttrs (_: spec: cfg.edid.applyWithKernelParameters && spec.edid != null))
           (lib.mapAttrsToList (output: spec: "${output}:edid/${spec.edid}"))
           (builtins.concatStringsSep ",")
           (p: lib.optional (p != "") "drm.edid_firmware=${p}")
         ]
       ;
     }
+    (
+      let
+        edids = lib.attrsets.filterAttrs (connector: output: cfg.edid.applyAtRuntime && output.edid != null) cfg.outputs;
+        loads = lib.pipe edids [
+          (lib.attrsets.mapAttrsToList (connector: output: "load ${connector} ${output.edid}"))
+          (builtins.concatStringsSep "\n")
+        ];
+      in
+      lib.mkIf (edids != { }) {
+        systemd.services.display-edid-apply = {
+          before = [ "graphical.target" "multi-user.target" ];
+          wantedBy = [ "graphical.target" "multi-user.target" ];
+          enable = true;
+          script = ''
+            export PATH="${lib.makeBinPath (with pkgs; [ findutils coreutils ])}:$PATH"
+
+            edid_paths=(
+              /run/current-system/firmware/edid
```

Comment:
```
$ find /run/current-system/firmware/edid/ -maxdepth 1 -name edid.bin
/run/current-system/firmware/edid/edid.bin

$ find /run/current-system/firmware/edid -maxdepth 1 -name edid.bin

$ ls /run/current-system/firmware/edid
edid.bin
```

The paths in edid_paths have to end with a / for maxdepth 1 to work. I tested this with normal files and it is not true, but for some reason it is true with /run/current-system/firmware/edid. Some weirdness because of it being a symlink maybe.

As a result: 

Applying '' to '/sys/kernel/debug/dri/0000:03:00.0/DP-1' output
cat: '': No such file or directory

Reply:
added `/` at the end of each entry